main/pipeline/pipeline.cc

```diff
@@ -1259,4 +1261,45 @@ bool cacheTreesAndFiles(const core::GlobalState &gs, WorkerPool &workers, vector
     return written;
 }
 
+vector<ast::ParsedFile> autogenCacheFiles(const core::GlobalState &gs, string_view cachePath,
```

Comment:
I think a better name for this would be something like `autogenWriteCacheFile` or something similarly indicative.